### ISSUE TITLE
Extension methods across assemblies?

### ISSUE THREAD

**Issue by Akonyl** [NLua/NLua#145]
I recently started using NLua, and saw an earlier issue where extension method functionality was added to NLua. I was able to get this to work, but issues arise when I attempt to use extensions from a separate assembly.

The situation is: I have a solution which contains a "Specific" project which is built upon a "Framework" project (which I keep generic for use in multiple projects). I import both assemblies to a lua state and have a SomeClass defined in the Framework project, however if I then try to use SomeClass extensions defined in the Specific project I only get errors saying:
[string "chunk"]:1: attempt to call method 'ExtensionDebug' (a string value)

If I define the extension method within the Framework project it works fine, and similarly if I define a class in Specific with extensions in Specific it also works, the issue only arises when the class is in Framework and the extensions in Specific. Currently I've gotten around this by instead inheriting from SomeClass in the Specific project, but that solution seems a bit messier so I'd rather use extensions if possible.

**Comment by viniciusjarina** [comment]
This is odd..NLua should check all loaded assemblies for extension methods https://github.com/NLua/NLua/blob/b46fed03058fdf8cb128867888e04845e88b1325/Core/NLua/Extensions/GeneralExtensions.cs#L155

I will try later. Thank you

**Comment by Akonyl** [comment]
It looks like this was some misunderstanding on my part in regards to CLR and import statements. When I used the following (as it looked like was suggested when reading the nlua.org frontpage), things broke in the way I described:
state.LoadCLRPackage();
state.DoString(@" import ('NLuaTestFramework.dll', 'NLuaTestFramework')");
state.DoString(@" import ('NLuaTestSpecific.exe', 'NLuaTestSpecific')");

However, when I changed the import statements to omit the assemblies, everything worked:
state.LoadCLRPackage();
state.DoString(@"import('NLuaTestFramework')");
state.DoString(@"import('NLuaTestSpecific')");

What's the purpose of the first form of import statement? I had thought that the first was equivalent to the second but it also imported the assembly, but from the looks of things that's done by loadCLRpackage anyway and it doesn't even manage the same functionality as the single-argument version.
